sui/gas-service.js

```diff
@@ -0,0 +1,251 @@
+            tx.pure.address(receiver), // Refund address
+            tx.pure.u64(atomicAmount), // Amount
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Gas refunded', receipt.digest);
+}
+
+async function processCommand(command, config, chain, args, options) {
+    const [keypair, client] = getWallet(chain, options);
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    if (!chain.contracts.axelar_gas_service) {
+        throw new Error('Axelar gas service contract not found');
+    }
+
+    switch (command) {
+        case 'pay_gas':
+            printInfo('Action', 'Pay gas');
+            await payGas(config, chain, args, options);
+            break;
+        case 'add_gas':
+            printInfo('Action', 'Add gas');
+            await addGas(config, chain, args, options);
+            break;
+        case 'collect_gas':
+            printInfo('Action', 'Collect gas');
+            await collectGas(config, chain, args, options);
+            break;
+        case 'refund':
+            printInfo('Action', 'Refund gas');
+            await refund(config, chain, args, options);
+            break;
+    }
+}
+
+async function mainProcessor(command, options, args, processor) {
+    const config = loadSuiConfig(options.env);
+    await processor(command, config, config.sui, args, options);
+    saveConfig(config, options.env);
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('gas-service').description('Interact with the gas service contract.');
+
+    const payGasProgram = new Command()
+        .command('pay_gas <amount> <destination_chain> <destination_address> <payload>')
+        .description('Pay gas for the new contract call.')
+        .requiredOption('--channel <channel>', 'Existing channel ID to initiate a cross-chain message over')
+        .option('--refund_address <refundAddress>', 'Refund address. Default is the sender address.')
+        .option('--params <params>', 'Params. Default is empty.')
+        .action((amount, destinationChain, destinationAddress, payload, options) => {
+            mainProcessor('pay_gas', options, [amount, destinationChain, destinationAddress, payload], processCommand);
+        });
+
+    const addGasProgram = new Command()
+        .command('add_gas <message_id> <amount>')
+        .description('Add gas for the existing contract call.')
+        .option('--refund_address <refundAddress>', 'Refund address.')
+        .option('--params <params>', 'Params. Default is empty.')
+        .action((messageId, amount, options) => {
+            mainProcessor('add_gas', options, [messageId, amount], processCommand);
+        });
+
+    const collectGasProgram = new Command()
+        .command('collect_gas <amount>')
+        .description('Collect gas from the gas service contract.')
+        .option('--receiver <receiver>', 'Receiver address. Default is the sender address.')
```

Comment:
this one should be optional since we can use sender address as a default receiver?

Reply:
ah, sure. we can make it required if people forget to supply it

Reply:
so, you think the receiver is better to always be specified? I thought the sender will be the receiver most of the time because you need to setup the right wallet that has gas collector cap.

Reply:
sorry, i meant we can wait and see. i'm ok with the sender as default for now